source/process/gitlab-process.rst

```diff
@@ -11,33 +11,133 @@ Merge Requests
+2. Before each merge request to GitLab Omnibus, upgrade is tested `following the steps below <https://docs.mattermost.com/process/gitlab-process.html#testing-upgrade-process>`_, using the `nightly Omnibus packages <https://packages.gitlab.com/gitlab/nightly-builds>`_ to validate the integration. This is so the packaging code and OAuth setup can be tested as well, which has historically been the main source of issues. Other test areas include:
 
  - Pre-provisioning OAuth configuration automatically on the Omnibus package
  - Mattermost ChatOps slash command integration
  - OAuth team creation option in GitLab Omnibus
 
+Upgrade Process
+~~~~~~~~~~~~~~~~~~
+
+Follow these steps to test the upgrade process for Mattermost in GitLab Omnibus.
+
+The current test servers are located at:
+
+ - `http://gitlab-rc-testing.spinmint.com <http://gitlab-rc-testing.spinmint.com>`_ - The GitLab instance itself
+ - `http://gitlab-rc-testing2.spinmint.com <http://gitlab-rc-testing2.spinmint.com>`_ - GitLab Mattermost
+
+The root admin account for them has username `root` and password `Password1`.
+
+1. Connect to the previously configured instance using the key you used to set it up. The host name is ``ubuntu@gitlab-rc-testing.spinmint.com``.
+2. Make sure your package manager is configured to allow for Gitlab RCs:
+
+ - Go to `https://packages.gitlab.com/gitlab/unstable <https://packages.gitlab.com/gitlab/unstable>`_.
+ - Click on the newest release candidate that’s marked CE and is for Ubuntu/Trusty.
+ - Run the first command listed on that page on your server. It looks something like:
+
+   .. code-block:: text
+
+     curl -s https://packages.gitlab.com/install/repositories/gitlab/unstable/script.deb.sh | sudo bash
+
+3. Update the package manager
+
+  .. code-block:: text
+
+    sudo apt-get update
+
+4. Update GitLab by running the second command listed on the page you opened on step 2. It looks something like:
+
+   .. code-block:: text
+
+     sudo apt-get install gitlab-ce=8.10.0-rc13.ce.0
+
+5. Reconfigure GitLab:
+
+  .. code-block:: text
+
+   sudo gitlab-ctl reconfigure
+
+6. Restart GitLab:
+
+  .. code-block:: text
+
+   sudo gitlab-ctl restart
+
+7. To confirm the upgrade was successful:
+
+ - Go to `http://gitlab-rc-testing2.spinmint.com  <http://gitlab-rc-testing2.spinmint.com >`_.
+ - Create an account and log in.
+ - Confirm the correct version number in **Main Menu** > **About Mattermost**.
+
+Useful Commands
+~~~~~~~~~~~~~~~~~~
+
+1. View config.json directly (arrow keys to scroll, press Q to exit):
+
+  .. code-block:: text
+
+    sudo less /var/opt/gitlab/mattermost/config.json
+
+2. Stop/start Mattermost manually:
+
+  .. code-block:: text
+
+	sudo /opt/gitlab/bin/gitlab-ctl stop mattermost
+	sudo /opt/gitlab/bin/gitlab-ctl start mattermost
+
+3. Access the GitLab admin console (press CTRL+D to exit):
+
+  .. code-block:: text
+
+	sudo gitlab-rails console production
+
+  You can then make actions such as update a user's password:
```

Comment:
For better readability, should this be "You can then carry out commands such as updating a user's password:" ?

Reply:
Thanks @lindy65! Updated